options.js

```diff
@@ -23,6 +23,11 @@ var tcDefaults = {
     vine.co
     imgur.com
     teams.microsoft.com
+  `.replace(regStrip, ""),
+  blacklistrc: `\
+    twitch.tv
```

Comment:
Are we sure about this? It seems like they have a good reason to increase rate (to catch up the stream) and we'd break that default behavior. It's not a bug on our part that the video speed jumps to 1.03.. it's exactly what twitch player intends.

Reply:
I thought I was sure, but now I'm not so sure anymore...

In my testing I never experienced the playing changing to 1.03x, I only experienced it bouncing back to 1.0x after attempting to change like shown in the gif in #584

Reply:
After fairly extensive testing with this fix disabled, I never experienced twitch.tv adjusting the playback rate of the video.

The only time I experienced an adjustment was when the playback buffer went beyond 30seconds, it will occasionally skip to a more recent time.

![image](https://user-images.githubusercontent.com/15102816/75684558-a0a3ec00-5c5e-11ea-9323-809ff56bf2e2.png)

What I suspect may be going on is some kind of adverse effect of using VSC without the fix applied. The original poster did explain the speed would never be fixed from 1.03 causing him to reload the entire page to fix. I would expect if this was intended behavior, it would auto-resolve once the playback has caught back up with the buffer.

Reply:
This is odd, it happened quickly when I tested again using the normal production version of the addon found in the Chrome store... and it did adjust back as you mentioned, so it felt like intentional behavior.

![image](https://user-images.githubusercontent.com/15102816/75686272-9800e500-5c61-11ea-9b62-98e0a6018674.png)

I'm now left wondering if something else in the updates was impacting this, or if I just got unlucky with my first tests. I'll keep poking at it and see what I can find out.

Reply:
Sorry for my monologue, found this one interesting. Either I messed up my tests the first go-around, or what I think is happening is this feature appears to be present on some streams but not others.

Visiting the same stream using the test environment, I was able to immediately see the speed change to 1.03 as it had in the production version of VSC. I didn't make any code or configuration changes between these tests. I suspect we may have stumbled upon some AB testing twitch is doing to see if this feature is worth/ready to push out to the entire user base.

![image](https://user-images.githubusercontent.com/15102816/75687308-39d50180-5c63-11ea-82ee-8bb1c9108874.png)

Knowing what I know now, I have a few thoughts.

First, we are already interfering with the default twitch playback rate - which may have been why they in turn be why they created a more aggressive mechanism for adjusting speed.

This is done on [line 149 of inject.js](https://github.com/igrigorik/videospeed/blob/master/inject.js#L149)

This has been a part of the behavior [since the initial commit](https://github.com/igrigorik/videospeed/blame/f1827ba63060255ba119431a00d6a26964058f84/inject.js#L61)

Second thought, because the fix does not actually block a ratechange from happening, this fix does not prevent the default behavior of twitch.tv. I validated this by adding a console.log statement before blocking the propagation

```
  if (isRateChangeBlocked()) {
    document.body.addEventListener(
      "ratechange",
      function(event) {
        console.log(
          "Speed changed to: " +
            event.target.playbackRate +
            " (event propogation blocked)"
        );
        event.stopImmediatePropagation();
      },
      true
    );
  }
```

With all of that known, I'm not sure which direction to go. I two valid paths we could take
1. Add twitch.tv to the blacklist as a default, and users are able to move it to the ratechange blocklist if they so desire
2. Add twitch.tv to the ratechange blacklist as a default, and users will be able to use the extension as they expect, without a visual indication of the speed changing to 1.03 then back to 1.0x once the stream is at 1.5 seconds behind the broadcaster.

I suppose the first thing we need to figure out is if we would like to disable the behavior of manually setting the speed to 1.0x every time we initialize videos - which is totally unrelated to the proposed changes here.

Reply:
> Add twitch.tv to the ratechange blacklist as a default, and users will be able to use the extension as they expect, without a visual indication of the speed changing to 1.03 then back to 1.0x once the stream is at 1.5 seconds behind the broadcaster.

If we implement ratechange blacklist and add twitch by default, my expectation is that the above behavior would not happen: the global listener would catch and block any updates initiated by non VSC logic. Is that not so?

Reply:
Good news! I came up with a new method that nullifies the need for a blacklist and should work on any website without interfering!

I think this is a much better fix that addresses your concerns. The new logic works on a timer. When you explicitly change the speed using video speed controller, it will block on rate change events that aren't generated by VSC for a 1 second window. That 1 second window is then refreshed every time a speed adjustment is done in VSC. Once you have stopped adjusting the speed for 1 second, it resumes normal operation.

In my tests, this fully blocks the "bounce back" effect with no ill effect on the regular operation of websites and the fix is seamless.

Reply:
Your expectation is actually not correct because the rate change command does not originate from a rate change listener.

I have come up with an even better solution that nullifies the need for a rate change block list. See my comments below.